formvalidations.js

```diff
@@ -0,0 +1,159 @@
+	}
+
+	var mail = document.forms["regform"]["email"].value;
+	var inpMatchn = mail.match(/[a-z0-9._+-]+@[a-z0-9]+\.[a-z]+/g);
+	if (inpMatchn != mail) {
+
+		alert("Enter Valid Email");
+		return false;
+	}
+
+	var pass = document.forms["regform"]["password"].value;
+	var inpMatch4 = pass.match(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*]).{8,}$/g);
+	if (inpMatch4 != pass) {
+		alert("Password should contain One Digit One SmallCase One UpperCase One from !@#$%^&*  and Must Be 8 characters long");
+		return false;
+	}
+
+	var cpass = document.forms["regform"]["confirmpassword"].value;
+	if (cpass != pass) {
+		alert("Passwords dont match");
+		return false;
+	}
+
+	return true;
+
+
+}
+
+function funcAddress() {
+
+
+	var homeadd = document.forms["regform"]["homeaddress"].value;
+	offadd = document.getElementById("compadd");
+	offadd.value = homeadd;
+
+
+}
+
+
+function checkingForm() {
+	if (validate() == true) {
+		var passcopy = document.getElementById('submit');
+		passcopy.removeAttribute("disabled");
+	}
+}
+
+
+function registering(forrm) {
+
+	if (validate(forrm) == true) {
+		alert("Registered");
+		var fname = forrm.firstname.value;
+		console.log(fname);
+		var lname = forrm.lastname.value;
+		var ha = forrm.homeaddress.value;
+		var oa = forrm.officeaddress.value;
+		var ml = forrm.email;
+		var mob = forrm.mobilephone;
+		var pass = forrm.password;
+		var cpass = forrm.confirmpassword;
+		var dash = document.getElementById("dashboard");
+		// var parainput = document.createElement("div");
+		// parainput.setAttribute("type", "button");
+		// parainput.setAttribute("value", fname);
+		var parainput = document.createElement("li");
+		var textnode = document.createTextNode(fname + " " + lname);
+		parainput.appendChild(textnode);
+		parainput.setAttribute("onmouseover", "mOver(this)");
+		parainput.setAttribute("class", "body-dashboard-eles");
+		dash.appendChild(parainput);
+	}
+
+}
+
+
+
+function mOver(obj) {
+	var mail = document.forms["regform"]["email"].value;
+	obj.title = mail;
+}
+
+
+
+function dropfunction() {
+	document.getElementById("myDropdown").classList.toggle("show");
+}
+
+
+window.onclick = function(event) {
+	if (!event.target.matches('.header-list2-name')) {
+
+		var dropdowntotals = document.getElementsByClassName("dropdown-content");
+		var i;
+		for (i = 0; i < dropdowntotals.length; i++) {
+			var openDropdown = dropdowntotals[i];
+			if (openDropdown.classList.contains('show')) {
+				openDropdown.classList.remove('show');
+			}
+		}
+	}
+}
+
+
+
+window.onload = function() {
+	var passpaste = document.getElementById('pass');
+	passpaste.onpaste = function(e) {
+		alert("Paste not allowed");
+		e.preventDefault();
+	}
+	var passpaste2 = document.getElementById('cpass');
+	passpaste2.onpaste = function(g) {
+		alert("Paste not allowed");
+		g.preventDefault();
+	}
+	passpaste.oncopy = function(h) {
+
+		h.preventDefault();
+	}
+	passpaste2.oncopy = function(u) {
+
+		u.preventDefault();
+	}
+
+}
+
```

Comment:
overall code looks good but you have applied the validation and submit button enabling logic on confirm password input only, u should have corresponding inputs validation running on its blur and u can take a boolean variables (flags) for all the inputs which u can be set to true or false depending on its validation test and if all flags are true (all input validation flags) then only enable the submit button.